frontend/src/components/RecipeDraft.js

```diff
@@ -1,16 +1,52 @@
-import {Button, makeStyles, TextField} from "@material-ui/core";
+import {Button, IconButton, makeStyles, TextField} from "@material-ui/core";
 import {useState} from "react";
 import IngredientEditor from "./IngredientEditor";
+import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
+import {CheckOutlined} from "@material-ui/icons";
+import styled from "styled-components/macro";
 
-export default function RecipeDraft({recipe, onRecipeNameUpdated, onIngredientsUpdated,  saveDraft}) {
+export default function RecipeDraft({recipe, onRecipeNameUpdated, onIngredientsUpdate, saveDraft}) {
     const [ingredients, setIngredients] = useState(recipe.ingredients);
 
     const classes = useStyles();
 
     const onIngredientUpdated = (updatedIngredient) => {
         setIngredients(ingredients.map(ingredient =>
             ingredient.id === updatedIngredient.id ? updatedIngredient : ingredient))
-        onIngredientsUpdated(ingredients)
+        onIngredientsUpdate(ingredients.map(ingredient =>
+            ingredient.id === updatedIngredient.id ? updatedIngredient : ingredient))
+    }
+
+    const onAddIngredient = () => {
+        const id = uuid();
+        setIngredients(
+            [...ingredients, {
+                "id": `${id}`,
+                "name": "",
+                "measurementUnit": "",
+                "amount": ""
+            }]
+        )
+        onIngredientsUpdate(
+            [...ingredients, {
+            "id": `${id}`,
+            "name": "",
+            "measurementUnit": "",
+            "amount": ""
+        }])
+    }
+
+    const onDeleteIngredient = (ingredientId) => {
```

Comment:
this also seems the function is running twice. What is the reason?

Reply:
the first one is updating the local state, while the second is updating the whole draft object that is tracked at another place, due to how react works I had to do it this way. (At least this way works as I intend to and I don't know any other way at the moment)